### PR TITLE
EOS-23135: Passing invalid interval parameter in the GET stats query returns 200

### PR DESCRIPTION
**Pull request by srikant03**
# Backend

# Problem Statement

EOS-23135: Passing invalid interval parameter in the GET stats query returns 200

- _Overview: JIRA number/GitHub Issue added to PR: [Y/N]:_Y
- _Describe the problem this patch intends to solve._
Passing invalid interval parameter in the GET stats query returns 200, it should return 400 response code.

## Design

- _For Bug describe the fix here._
- _For feature, post the link to the solution page._
- _Any northbound interface change and has been notified to other gatekeepers [Y/N]:_

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_ Yes
- _Confirm All CODACY errors are resolved. Yes/No?_ Yes

## Testing

- _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No?_ No
- _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No?_ No
- _Confirm Testing was performed with installed RPM. Yes/No?_ No
- _Confirm all the ut/regression/smoke test has been performed [Y/N]_ Yes

## Checklist

- _PR is self-reviewed? Yes/No?_ Yes
- _GitHub Issue is updated_ 
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_ No
        - _Change notified to other gatekeepers: Yes/No?_
        - _Code reviews done and addressed: Yes/No?_ Yes
        - _Codacy issues reviewed and addressed: Yes/No?_ Yes
        - _Deployment test : Done/Not?_ Not
        - _UT/smoke test: Done/Not?_ Done
        - _Jira number added to PR: Yes/No? Yes_
        - _Github merge done using UI: Yes/No?_ No
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by srikant03** [comment]
interval1

interval2

File: csm/common/timeseries.py [hunk 1]
```diff
@@ -270,15 +270,20 @@ async def _parse_interval(self, from_t, duration_t, interval, total_sample):
         duration_t = int(duration_t) - int(self._offset_interval)
         if total_sample == "" and interval == "":
             interval = str(self._storage_interval) + 's'
-        elif total_sample != "":
+        elif total_sample != "" and interval == "":
```
Comment (pawankumarsrivastava) [inline csm/common/timeseries.py:273]: This check is not necessary here -- > interval == "" . If total_sample is given then `interval` query parameter will not come into the picture, so any validation should be skipped .

File: csm/common/timeseries.py [hunk 2]
```diff
@@ -270,15 +270,20 @@ async def _parse_interval(self, from_t, duration_t, interval, total_sample):
...
-        else:
-            raise CsmInternalError("Unable to parse interval")
+            try:
+                interval = int(interval)
+            except ValueError:
+                raise InvalidRequest("'interval' should be integer and greater than zero or blank")
+            if interval < 1:
+                raise InvalidRequest("'interval' should be integer and greater than zero or blank")
+            else:
+                interval = str(int(interval)) + 's'
```
Comment (pawankumarsrivastava) [inline csm/common/timeseries.py:286]: If this statement fails return InvalidRequest error.

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

**Review by srikant03 — COMMENTED** [review]
Addressed all review comments.

File: csm/common/timeseries.py (hunk 1 above)
Comment (srikant03) [reply csm/common/timeseries.py]: Addressed

File: csm/common/timeseries.py (hunk 2 above)
Comment (srikant03) [reply csm/common/timeseries.py]: Addressed

Comment (pawankumarsrivastava) [reply csm/common/timeseries.py]: You can do like this - 
try:
	interval = int(interval)
	if interval < 1:
		raise InvalidRequest("'interval' should be integer and greater than zero.")
	interval = str(interval) + 's'
except (ValueError, TypeError):
	raise InvalidRequest("'interval' should be integer and greater than zero.")
   
Also, one thought if you are checking for lower value like -ve then you should check for higher value too. Like what if I give 99898988889998889998 as a value?
For now its OK.

Comment (srikant03) [reply csm/common/timeseries.py]: > You can do like this -
> try:
> interval = int(interval)
> if interval < 1:
> raise InvalidRequest("'interval' should be integer and greater than zero.")
> interval = str(interval) + 's'
> except (ValueError, TypeError):
> raise InvalidRequest("'interval' should be integer and greater than zero.")
> 
> Also, one thought if you are checking for lower value like -ve then you should check for higher value too. Like what if I give 99898988889998889998 as a value?
> For now its OK.

Addressed, I had asked to Pranay for Higher value, but he said that there is no higher value limit, that why did not check that.

**Review by kupranay — APPROVED** [review]

**Pull request merged**